### DISCUSSION TITLE
Kann man die scripte auch irgendwie für influxdb/grafana nutzen? also ohne home assistant?

### DISCUSSION THREAD

**Discussion by sreutter** [stephan192/hochwasserportal#32] [General]
Danke,
Stefan

**Comment by stephan192** [comment]
Wahrscheinlich. Die relevanten Funktionen sind alle im Unterordner **lhp_api** zusammengefasst.
Einfach eine Instanz der Klasse **HochwasserPortalAPI** mit einem gültigen Pegel instanzieren und dann die Werte mittels der Propertyfunktionen auslesen. Anschließend die Daten in der InfluxDB ablegen.

**Reply by sreutter** [reply]
Danke.
da ich leider nicht ganz so firm bin in python, wollte ich fragen ob es möglich wäre, etwas detailliertere Hinweise zu bekommen?

**Reply by sreutter** [reply]
P.S.:
ich habe gesehen, daß es für manche Pegel auch mehr als die 3 Werte gibt, z.b. für den Pegel Raunheim zusätzlich
Lufttemperatur [°C].Sauerstoffgehalt [mg/l],Wassertemperatur [°C]. Könnte man die auch noch einbauen?

**Comment by stephan192** [comment]
**Frage 1: Beispiel**
Ein einfaches Beispiel (unter der Annahme das folgendes Skript im Überordnuer zum Ordner lhp_api liegt):

```
from lhp_api import HochwasserPortalAPI

lhp = HochwasserPortalAPI("BY_16005701")

lhp.update()
print(lhp.level)
print(lhp.stage)
print(lhp.flow)
```

lhp.update() muss zyklisch (ca. alle 15min) aufgerufen werden.
Mit Influx habe ich noch nichts zu tun gehabt.

**Frage 2: Temperatur & Co.**
Ja, das ist prinziell möglich. Da es aber nur wenige Pegel unterstützen plane ich aktuell keine Unterstützung dieser Werte.

**Reply by sreutter** [reply]
Danke!
prinzipiell tut es:

./pegel_mainz.py
294.0
0
None

allerdings fehlt der Wert für den Abfluss. der ist auf der wsv-seite schon vorhanden. Eine Idee, warum der nicht mit ausgegeben wird? Ich versuche mal, mir das selbst ein wenig anzuschauen, aber ich rechne nicht damit, gleich den Durchblick zu kriegen...
https://pegelonline.wsv.de/gast/stammdaten?pegelnr=25100100

PS.: hab mich gewundert, warum der Pegel mit HE_25100100 angegeben wird. Mainz ist doch in Rheinland-Pfalz?
hätte jetzt RP_usw. erwartet?

**Comment by stephan192** [comment]
Das ist alles im README.md beschrieben. Bitte erst lesen und dann fragen.

Aber hier nochmal:
- Manche Pegel sind auf zwei Portalen gelistet, meistens wenn sie an der Grenze zwischen zwei Bundesländern liegen. Mainz gibt es auch mit RP_ vorne.
- Die Abfrage läuft über die Länderportale und nicht über pegelonline.wsv.de Was die Länderportale zur Verfügung stellen gibt es, dass was sie nicht zur Verfügung stellen gibt es nicht.

**Reply by sreutter** [reply]
sorry, das hatte ich überlesen mit den doppelten Pegeln, die Readme.md hab ich mir schon angeschaut.
wenn ich den Pegel mit RP vorne aufrufe, klappt auch das Auslesen vom Abfluss.
somit hab ich eigentlich fast alles, die übergabe an die influxdb ist dann schnell gemacht. 
was ich gerade noch nicht sehe/weiss, wie ich das Datum zum Pegel mit der aktuell gültigen Zeit abrufen kann. habe 
lhp.last_update probiert, aber das ist wohl in UTC? wie kann ich das in MEZ bekommen? bzw. MESZ, wenn es soweit ist?

Interessant: in dem json (also jedenfalls für den mainzer pegel) werden auch Vorhersagen mitgeliefert ("predictions").

Vielen Dank für das tolle Projekt!

**Reply by sreutter** [reply]
P.S.: scheint usus zu sein, die utc-zeit zu übergeben. somit hat sich das wohl erledigt.
was mir noch aufgefallen ist - der wert für den abfluss hinkt immer eine viertelstunde nach, hat also eigentlich einen anderen Zeitstempel.
(so sieht es jedenfalls auf dem wsv-portal aus).

**Reply by stephan192** [reply]
Ja, die Zeiten sind alle UTC und das ist Absicht.
Ich rate dringend davon ab die Zeit vor dem Abspeichern in der Datenbank in die lokale Zeit zu konvertieren. Die Konvertierung sollte immer erst im allerletzten Schritt (der Anzeige der Daten) erfolgen. Wenn man mal ein bisschen mit Zeiten gerechnet hab erklärt es sich schnell von selbst warum man Zeiten eigentlich immer nur in UTC abspeichert.

Ja, bei manchen Portalen gibt es für Abfluss und Pegel je einen Zeitstempel. Meine Integration und das LHP-Portal verweden aber nur einen und das ist wenn verfügbar der des Pegels.

**Comment by stephan192** [comment]
Hallo @sreutter inzwischen habe ich die API aus der Home Assistant Integration herausgelöst und als eigenes Paket bei PyPI hochgeladen.
https://pypi.org/project/lhpapi/

**Reply by sreutter** [reply]
ah, das ist interessant.
ich habe die bisherige Lösung jetzt schon länger laufen, funktioniert sehr gut (s. z.b. den Screenshot).
was ich noch machen möchte, ist weitere Pegel mit noch mehr Parametern einbinden, wie weiter oben beschrieben die Daten vom pegel Raunheim. vielleicht klappt das damit ja auch besser jetzt?
nochmals danke für die super Integration!
![grafik](https://github.com/stephan192/hochwasserportal/assets/12584679/ccf96036-6002-4289-b814-776e9af646ed)

**Comment by stephan192** [comment]
Die "neue" API entspricht in Ihrem Funktionsumfang 100% dem Code den Du bereits verwendest. Den Einbau neuer Features plane ich aktuell nicht.
